pkg/poddisruptionbudget/poddisruptionbudget.go

```diff
@@ -0,0 +1,254 @@
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Creating pod disruption budget %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+	if !builder.Exists() {
+		builder.Object, err = builder.apiClient.PodDisruptionBudgets(
+			builder.Definition.Namespace).Create(context.TODO(),
+			builder.Definition, metav1.CreateOptions{})
+	}
+
+	return builder, err
+}
+
+// Delete deletes the PodDisruptionBudget from the cluster.
+func (builder *Builder) Delete() error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof("Deleting pod disruption budget %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	if !builder.Exists() {
+		glog.V(100).Infof("Pod disruption budget %s cannot be deleted because it does not exist",
+			builder.Definition.Name)
+
+		builder.Object = nil
+
+		return nil
+	}
+
+	err := builder.apiClient.PodDisruptionBudgets(builder.Definition.Namespace).Delete(context.TODO(),
+		builder.Definition.Name, metav1.DeleteOptions{})
+
+	if err != nil {
+		return err
+	}
+
+	builder.Object = nil
+
+	return nil
+}
+
+// Exists checks if the PodDisruptionBudget exists in the cluster.
+func (builder *Builder) Exists() bool {
+	glog.V(100).Info("Checking if the PodDisruptionBudget exists in the cluster")
+
+	var err error
+	builder.Object, err = builder.apiClient.PodDisruptionBudgets(
+		builder.Definition.Namespace).Get(context.TODO(),
+		builder.Definition.Name,
+		metav1.GetOptions{})
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// WithPDBSpec sets the PodDisruptionBudgetSpec for the PodDisruptionBudget.
+func (builder *Builder) WithPDBSpec(spec policyv1.PodDisruptionBudgetSpec) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	builder.Definition.Spec = spec
+
+	return builder
+}
+
+// Update updates the PodDisruptionBudget in the cluster.
+func (builder *Builder) Update(force bool) (*Builder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Updating pod disruption budget %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	if !builder.Exists() {
+		return builder, fmt.Errorf("pod disruption budget %s does not exist in namespace %s",
+			builder.Definition.Name, builder.Definition.Namespace)
+	}
+
+	_, err := builder.apiClient.PodDisruptionBudgets(
+		builder.Definition.Namespace).Update(context.TODO(),
+		builder.Definition, metav1.UpdateOptions{})
+
+	if err != nil {
+		if force {
+			glog.V(100).Infof("Force updating pod disruption budget %s in namespace %s",
+				builder.Definition.Name, builder.Definition.Namespace)
+
+			err := builder.Delete()
+			if err != nil {
+				glog.V(100).Infof(msg.FailToUpdateError("pod disruption budget",
+					builder.Definition.Name, builder.Definition.Namespace))
+			}
+
+			return nil, err
+		}
+
+		return builder.Create()
```

Comment:
This return needs to be swapped with the one on line 214

Reply:
Done.

Reply:
@sebrandon1 I think these still need to be swapped so the `return builder.Create()` is inside the `if force`

Reply:
Okay I moved this and also the `resourcequotas.go` was wrong as well.